### PR TITLE
feat(module:notification): add onClick observable

### PR DESCRIPTION
**Pull request by wendellhu95**
refactor message and notification

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

`NzNotificationDataFilled` is replaced by `NzNotificationRef`.

`NzMessageDataFilled` is replaced by `NzMessageRef`.

Doing this to make these two components' API consistent with others', and to stop exposing some internal implementation details.

### CODE REVIEW

**Comment by hsuanxyz** [comment]
notification API 文档没改，还是 `NzNotificationDataFilled `, 另外这是 breaking change

File: components/message/doc/index.en-US.md [hunk 1]
```diff
@@ -57,12 +57,12 @@ You can use `NzConfigService` to configure this component globally. Please check
...
 
-### NzMessageDataFilled
+### NzMessageRef
 
 It's the object that returned when you call `NzMessageService.success` and others.
 
 ```ts
-export interface NzMessageDataFilled {
+export interface NzMessageRef {
   onClose: Subject<false>; // It would emit an event when the message is closed
```
Comment (hsuanxyz) [inline components/message/doc/index.en-US.md:66]: 这里类型是  `Subject<boolean>` 吗？还是 `Subject<false>`

File: components/notification/doc/index.zh-CN.md [hunk 2]
```diff
@@ -77,5 +77,6 @@ import { NzNotificationModule } from 'ng-zorro-antd/notification';
 ```ts
 export interface NzNotificationDataFilled {
   onClose: Subject<boolean>; // 当 notification 关闭时它会派发一个事件，如果为用户手动关闭会派发 `true`
+  onClick: Subject<MouseEvent>;
```
Comment (hsuanxyz) [inline components/notification/doc/index.zh-CN.md:80]: `onClose ` 是以前就有的，title 应该是 `feat(module:notification): add onClick observable` ?

Comment (wendellhu95) [reply components/notification/doc/index.zh-CN.md:81]: 改啦

File: components/message/doc/index.en-US.md (hunk 1 above)
Comment (wendellhu95) [reply components/message/doc/index.en-US.md:67]: message 暂时没有提供手动关闭的功能，所以目前是 `false`

File: components/message/typings.ts [hunk 3]
```diff
@@ -17,22 +17,20 @@ export interface NzMessageDataOptions {
...
   state?: 'enter' | 'leave';
+
   onClose?: Subject<boolean>;
 }
+
+export type NzMessageRef = Pick<Required<NzMessageData>, 'onClose' | 'messageId'>;
+
+/**
+ * @deprecated use `NzMessageRef` instead
+ */
```
Comment (hsuanxyz) [inline components/message/typings.ts:35]:
```suggestion
 * @breaking-change 10.0.0
 */
```

File: components/notification/typings.ts [hunk 4]
```diff
@@ -27,25 +19,30 @@ export interface NzNotificationDataOptions<T = {}> {
...
+  // observables exposed to users
+  onClose?: Subject<boolean>;
+  onClick?: Subject<MouseEvent>;
 }
+
+export type NzNotificationRef = Pick<Required<NzNotificationData>, 'onClose' | 'onClick' | 'messageId'>;
+
+/**
+ * @deprecated use `NzNotificationRef` instead
+ */
```
Comment (hsuanxyz) [inline components/notification/typings.ts:47]:
```suggestion
 * @breaking-change 10.0.0
 */
```

**Pull request merged**